lib/rules/prop-types.js

```diff
@@ -917,6 +917,12 @@ module.exports = {
         }
       },
 
+      TypeParameterInstantiation: function(node) {
+        if (node.params && node.params[0].type === 'GenericTypeAnnotation' && node.params[0].id.name === 'Props') {
```

Comment:
That makes sense.

But what about this test case that we have. Is that still valid? I suppose not, as now we are passing `void` for `Props` and ` { person: Person }` as `State` ? What is the third parameter? Context?

```js
type Person = {
  firstname: string
};
class Hello extends React.Component<void, { person: Person }, void> {
  render () {
    return <div>Hello {this.props.person.lastname}</div>;
  }
}
```